spec/unit/widgets/textbox_spec.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+require 'rspec'
+
+module LibyuiClient
+  module Widgets
+    RSpec.describe Textbox do
+      let(:widget_controller) { instance_double('widget_controller') }
+      let(:widget_filter) { instance_double('Filter', { id: 'aliases' }) }
+      subject { Textbox.new(widget_controller, widget_filter) }
+      describe '#set' do
+        it 'can send POST with new value' do
+          allow(widget_controller).to receive(:send_action)
+          allow(widget_controller).to receive(:send_action)
+          expect(subject).to receive(:action).with({ action: 'enter_text', value: 'test' })
+          subject.set('test')
+        end
+      end
+      # of #set
+
+      describe '#value' do
+        it 'should return the text of the :value' do
+          response = double('Response', { body: [{ 'class': 'InputField',
```

Comment:
as discussed I think we don't need to worry about the response here, just that property is called properly and return the value expected, for example if developer changes `:value` by other field by accident, it will be detected:
```ruby
 it 'calls property for value field and return its value' do
    expect(textbox).to receive(:property).with(:value).and_return(value)
    expect(textbox.value).to eq(value)
end
```

Reply:
the thing is that we dont test anything in this case and the test will always pass. with the response we make sure that when the property method of the base class doesnt break the value method of the child class. I will leave it as it is. Maybe it is not even needed and i can be removed.